apps/strapi-cms/src/api/webinar/content-types/webinar/lifecycles.ts

```diff
@@ -0,0 +1,41 @@
+// The only way to make this code work is to do side effects, and throwing exceptions and creating functions
+/*eslint-disable*/
+import { errors } from '@strapi/utils';
+
+interface IWebinarEvent {
+  readonly params: {
+    readonly data: {
+      readonly startDatetime?: string;
+      readonly endDatetime?: string;
+    };
+  };
+}
+
+const validateDates = (event: IWebinarEvent) => {
+  const { data } = event.params;
+
+  const startDateTime = data.startDatetime
+    ? new Date(data.startDatetime)
+    : null;
+  const endDateTime = data.endDatetime ? new Date(data.endDatetime) : null;
+
+  if ((startDateTime && !endDateTime) || (!startDateTime && endDateTime)) {
+    throw new errors.ApplicationError(
+      'Both start and end dates must be provided, or none should be set'
+    );
+  }
+
+  if (startDateTime && endDateTime && endDateTime < startDateTime) {
+    throw new errors.ApplicationError('End date must be after start date');
+  }
+};
+
+module.exports = {
+  beforeCreate(event: IWebinarEvent) {
+    validateDates(event);
+  },
+
+  beforeUpdate(event: IWebinarEvent) {
+    validateDates(event);
+  },
+};
```

Comment:
```suggestion
// The only way to make this code work is to do side effects, throwing exceptions and creating functions
import { errors } from '@strapi/utils';

interface IWebinarEvent {
  readonly params: {
    readonly data: {
      readonly startDatetime?: string;
      readonly endDatetime?: string;
    };
  };
}

// eslint-disable-next-line functional/no-return-void,@typescript-eslint/explicit-function-return-type
const validateDates = (event: IWebinarEvent) => {
  const { data } = event.params;

  const startDateTime = data.startDatetime
    ? new Date(data.startDatetime)
    : null;
  const endDateTime = data.endDatetime ? new Date(data.endDatetime) : null;

  if ((startDateTime && !endDateTime) || (!startDateTime && endDateTime)) {
    // eslint-disable-next-line functional/no-throw-statements
    throw new errors.ApplicationError(
      'Both start and end dates must be provided'
    );
  }

  if (startDateTime && endDateTime && endDateTime < startDateTime) {
    // eslint-disable-next-line functional/no-throw-statements
    throw new errors.ApplicationError('End date must be after start date');
  }
};

// eslint-disable-next-line functional/immutable-data,functional/no-expression-statements
module.exports = {
  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type,functional/no-return-void,prefer-arrow/prefer-arrow-functions
  beforeCreate(event: IWebinarEvent) {
    // eslint-disable-next-line functional/no-expression-statements
    validateDates(event);
  },

  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type,functional/no-return-void,prefer-arrow/prefer-arrow-functions
  beforeUpdate(event: IWebinarEvent) {
    // eslint-disable-next-line functional/no-expression-statements
    validateDates(event);
  },
};
```

Reply:
@tommaso1 @marcobottaro I added [this task](https://pagopa.atlassian.net/browse/DEV-1594) to update linting for strapi-cms app to align linting configuration and avoid all these eslint disable in future